test/test_viz.py

```diff
@@ -0,0 +1,48 @@
+import jax.numpy as jnp
+import pygame
+import pygame.locals
+
+from cotix._bodies import Ball
+from cotix._visualizer_tools import BallDrawer, Visualizer
+
+
+def test_viz():
+    framerate = 60
+
+    viz = Visualizer(100, 6, 6, name="Test Viz")
+
+    b = Ball.make_default()
+
+    b = b.set_position(b.position + jnp.array([3.0, 3.0]))
+
+    print(b.position)
+
+    b_drawer = BallDrawer(b, pygame.Color(255, 0, 0))
+
+    run = True
+    framecount = 0
+    inc = jnp.array([0.1, 0.1])
+
+    while run:
+        # Iterating over all the events received from
+        # pygame.event.get()
+        for event in viz.get_events():
+            # If the type of the event is quit
+            # then setting the run variable to false
+            if event.type == pygame.locals.QUIT:
+                run = False
+        # Draws the surface object to the screen.
+
+        viz.draw([b_drawer])
+
+        pygame.time.Clock().tick(framerate)
+
+        framecount += 1
+
+        # update the ball
+        b = b.set_position(b.position + inc)
+        # update the drawer with the ball
+        b_drawer.object = b
+
+        if framecount == framerate:
+            run = False
```

Comment:
Seems like a bad way to write condition. why exatcly this value? 

And, besides, you still don't make any assertions in the test: it just checks that you have pygame installed. Maybe you can think about some meaningful property that you can test, like, that you actually draw something? Or just remove the test completely.

If you want to test that some functionality works, like that it draws something, don't commit these files: these tests are meaningless for other people, so they should not be shared

Reply:
As I want the simulation to run over 1s, the number of frames executed will be equal to the framerate you're achieving.

Thinking about a way to test with assertions, we could run the viz, save the image and then comparing future tests simulations to this image. In that way, we can test that everything is drawn in the same way as our "correct" run

Reply:
Yeah, this does make sense, but also introduces a lot of internal dependecies: imagine we change collision resolution 'a little bit', and, while simulation is still physically plausible, your tests started failing.

I think we should not test the visualizer in any meaningful way for now: while it makes sense to test it, the complexity of tests would significantly exceed the complexity of the visualizer itself, which means it is not worth the effort.

Reply:
I understand. However, it would be cool to have something like a "visualizer demo"  to try out how the drawing for different objects work. Should we include this in another folder `demos`?

Reply:
Nope, end users don't care about how cool our visualizer is, they care that it works.

But I like the idea of having a folder like "examples", though its still not really what you mean. So id suggest that we don't do that until implementation of environments is ready

Reply:
I mean, look, more (good written) code is never worse than less. But implementing demos looks useless to me. Why would somebody who is trying to get his PPO, or whatever RL thing, working, care that visualizer can show collisions? They care only about the fact that it looks like the agent is playing well. I mean, generally, I think the standard way of determining that your model works is judging by metrics, and in our case we need a visualization only because we don't know what reward quantifies sufficiently good performance: so visualizer allow us to see if the agent can perform in alignment with our expectations of what good behaviour is, sort of a way to do human feedback. And I don't think that information about these collisions/balls catching robots is critical for judging about the overall performance of the agent. Yes, this informatino is useful for debugging, and overall for understanding why agent acts the way it does, but it is not critical, at least on the current stage.

Again, feel free to implement any of the two approaches, since both of them are sufficiently good to start with. I don't have a preference between the two anymore.